### PR TITLE
Standardize obj/updateicon() and update_icons() to update_icon()

### PR DESCRIPTION
**Pull request by SpaceManiac**
Not good for `O.update_icon()` to do nothing because you need to call `O.updateicon()` instead

### CODE REVIEW

**Comment by MrPerson** [comment]
This kind of stuff is really, really good

File: code/game/objects/items/pneumaticCannon.dm [hunk 1]
```diff
@@ -243,9 +243,9 @@
 			return
 		to_chat(user, "<span class='notice'>You hook \the [thetank] up to \the [src].</span>")
 		tank = thetank
-	update_icons()
+	update_icon()
 
-/obj/item/pneumatic_cannon/proc/update_icons()
+/obj/item/pneumatic_cannon/update_icon()
```
Comment (Fox-McCloud) [inline code/game/objects/items/pneumaticCannon.dm]: this is an infinite loop! Be careful!

**Comment by Fox-McCloud** [comment]
mobs also use `update_icons`

**Comment by SpaceManiac** [comment]
> `This is an infinite loop! Be careful!`

Ugh why was it like this

> Why not just make update_icon an atom level proc and make mobs use it too?

Future PR

**Pull request merged**